### ISSUE TITLE
Extract and migrate to use an image similarity GStreamer Plugin

### ISSUE THREAD

**Issue by rafaelcaricio** [cbsinteractive/hawkeye#39]
The part of Hawkeye that captures the buffer from the pipeline and matches it to a slate image could be extracted into a standalone GStreamer plugin. We can maybe get some inspiration from the [SSIM validate plugin](https://gstreamer.freedesktop.org/documentation/gst-devtools/plugins/ssim.html?gi-language=c). The difference is that we want our plugin to be used in a live pipeline and somehow hook callbacks when the current buffer image ranks above the ssim value we provide.

One advantage of having this part of the code as a plugin is that we can easily test multiple configurations and use the plugin other applications.

**Comment by LyleScott** [comment]
I noticed there was also https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad/html/gst-plugins-bad-plugins-IQA.html which actually uses our rust DSSIM lib.

**Comment by LyleScott** [comment]
I think it's a bit tougher to do this task given the arrival of the multi slate feature where a single Watcher can have >=1 slates to match against and the slate being at the `transition` level instead of being on the root object for the whole watcher. 

I'm still quite unfamiliar with the domain and plugins, so maybe it just __feels__ harder.

**Comment by rafaelcaricio** [comment]
The referenced plugin does not expose a callback that we can hook to execute if detection passes a threshold. We would need to write a plugin specific to our use case.

The plugin would just make a comparison of images, how many images we compare does not matter. We just need to instantiate one plugin per image we want to compare. We would adapt the GStreamer pipeline to have on branch for each slate we want to detect.

In order to implement this task the way it is described, you need to get quite familiar with GStreamer and the Rust API it provides. I would suggest going through the GStreamer tutorials in the [gstreamer-rs](https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/tree/master/tutorials/src/bin) repo, the explanations for each part in https://gstreamer.freedesktop.org/documentation/tutorials/basic/hello-world.html?gi-language=c .

In short: This requires a good understanding of GStreamer.

**Comment by rafaelcaricio** [comment]
I've created the `videocompare` element, and it is now merged to upstream to GStreamer https://gstreamer.pages.freedesktop.org/-/gst-plugins-rs/-/jobs/29941220/artifacts/documentation/videofx/videocompare.html?gi-language=c
